### PR TITLE
A new PR for refactoring

### PR DESCRIPTION
**Pull request by StephenDsouza90**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+Employee/Customer - methods that stores objects
+
+### core.py
+
+Core contains the controller class that inherites from the SQLiteBackend class and composites from the Employee and Customer classes.
+
+The controller class also includes all the methods from the Employee and Customer.
+
+The purpose of the controller class is to mainly connect to the SQLiteBackend class and use the session as well as link the Employee and Customer classes to perform the functions.
+
```
Comment (dsouzajude) [inline README.md:110]: Also create a separate heading called:

**How to run locally:**

```bash

>> python main.py

```

And then show a few sample inputs like on how to add food categories, just copy and paste the whole thing from the command prompt.

Comment (StephenDsouza90) [reply README.md:110]: change made

File: README.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+Core contains the controller class which inherites from the SQLiteBackend class and composites from the Employee and Customer classes.
+
+The controller class also includes all the methods from the Employee and Customer.
+
+The purpose of the controller class is to mainly connect to the SQLiteBackend class and use the session as well as link the Employee and Customer classes to perform the functions.
+
+## How to run locally:
+
+>> python main.py
```
Comment (dsouzajude) [inline README.md:42]: This is not correct formatting.

Try to wrap it around: \```bash and \```

\```bash

\>> python main.py

\```

Comment (StephenDsouza90) [reply README.md:42]: change made

File: core.py [hunk 3]
```diff
@@ -140,66 +121,61 @@ def process_order(self, cust_id):
...
-                print("Items {}".format("updated" if True else "not updated"))
+                u = self.customer.update_food_to_order(session, o.order_id, food_id, food_qty)
+                print("Items {}".format("updated" if u else "not updated"))
 
             option = int(input(selection))
         
         if o:
-            print("Order {}. \nOrder number: {}".format("generated", o.order_id if True else "not generated"))
-
+            print("Order {}. \nOrder number: {}".format("generated", o.order_id if o else "not generated"))
```
Comment (dsouzajude) [inline core.py]: This doesn't make sense. Why are you checking `if o` twice?

File: core.py [hunk 4]
```diff
@@ -140,66 +121,61 @@ def process_order(self, cust_id):
...
-
-    def cancel_order(self, order_id, order_status):
-        session = self.Session()
-        c = self.customer.cancel_order(order_id, order_status, session) 
+    @handle_session
+    def cancel_order(self, session, order_id, order_status):
+        c = self.customer.cancel_order(session, order_id, order_status) 
         if c:
-            print("Cancel {}!".format("successful" if True else "not successful"))
+            print("Cancel {}!".format("successful" if c else "not successful"))
```
Comment (dsouzajude) [inline core.py]: Again, you're checking `if c` twice. It's wrong.

File: models.py [hunk 5]
```diff
@@ -105,162 +122,94 @@ class CustOrderStatus(Base):
...
+        return result
+            
+    def sum_revenue_today(self, session, order_status):
+        """ Sum revenue/sales of today """
+
+        rev = text("""SELECT sum(cos.bill_amount) 
+            FROM customer_order_status as cos 
+            WHERE cos.order_status = {} AND date(cos.checkout_time) = date(CURRENT_DATE);""".format(order_status))
+        result = session.connection().execute(rev).fetchall()
+        return result
```
Comment (dsouzajude) [inline models.py:178]: What's the difference b/w `view_revenue_today(...)` and `sum_revenue_today(...)`?

File: models.py [hunk 6]
```diff
@@ -420,50 +305,29 @@ def checkout(self, order_id, order_status, order_address, checkout_time, estimat
...
-        finally:
-            session.expunge_all()
-            session.close()
-
-
-    def view_orders_status(self, order_id, session):
+        session.commit()
+        return update
+    
+    def view_orders_status(self, session, order_id):
```
Comment (dsouzajude) [inline models.py]: This function is repeated twice for employee and customer. It's code repetition! It's not good. So you can actually create a function called `view_order_status()` which is not part of any class and then call it in the Employee and Customer class.

File: core.py (hunk 3 above)
Comment (StephenDsouza90) [reply core.py]: changed made

File: core.py (hunk 4 above)
Comment (StephenDsouza90) [reply core.py]: change made

File: models.py (hunk 5 above)
Comment (StephenDsouza90) [reply models.py:178]: changed view_revenue_today() to view_sales_today()

File: models.py (hunk 6 above)
Comment (StephenDsouza90) [reply models.py]: created a class for the common functions that the employee and customer can use.

File: models.py [hunk 7]
```diff
@@ -484,12 +291,38 @@ def update_order(self, order_id, order_status, session):
...
-            raise Exception("update not completed!")
-        finally:
-            session.expunge_all()
-            session.close()
\ No newline at end of file
+        session.commit()
+        return update
+
+
+class CommonFunctions:
```
Comment (dsouzajude) [inline models.py]: You don't need to create a class for the common functions. Just create normal functions.

Comment (StephenDsouza90) [reply models.py]: created normal functions

File: core.py (hunk 3 above)
Comment (dsouzajude) [inline core.py]: Don't check for `o` here, just print it. Check for `o` on top. On top, do like this:

```python
if not o:
   print("...bla bla o not successful...")
   return
```

File: core.py [hunk 8]
```diff
@@ -1,129 +1,110 @@
...
+        login = self.customer.customer_login(session, cust_id)
         return login
 
-
-    def process_order(self, cust_id):
-        session = self.Session()
-        o = self.customer.create_order_id(cust_id, session)
+    @handle_session
+    def process_order(self, session, cust_id):
+        o = self.customer.create_order_id(session, cust_id)
```
Comment (dsouzajude) [inline core.py:82]: Check for `o` here:

```python
if not o:
   print("...bla bla o not successful...")
  return
```

**Review by dsouzajude — APPROVED** [review]
Nice job! Good work. Looks good to me now. Partayyy!

**Pull request merged**